src/core/ddsc/src/dds__sysdef_model.h

```diff
@@ -0,0 +1,706 @@
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_DATA_FRAME_SPEC_IPV6_TUPLE,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_DATA_FRAME_SPEC_IPV6_TUPLE_SOURCE_IP_ADDRESS,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_DATA_FRAME_SPEC_IPV6_TUPLE_DESTINATION_IP_ADDRESS,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_DATA_FRAME_SPEC_IPV6_TUPLE_DSCP,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_DATA_FRAME_SPEC_IPV6_TUPLE_PROTOCOL,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_DATA_FRAME_SPEC_IPV6_TUPLE_SOURCE_PORT,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_DATA_FRAME_SPEC_IPV6_TUPLE_DESTINATION_PORT,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_PERIODICITY,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_SAMPLES_PER_PERIOD,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_MAX_BYTES_PER_SAMPLE,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_TRANSMISSION_SELECTION,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_TIME_AWARE,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_TIME_AWARE_EARLIEST_TRANSMIT_OFFSET,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_TIME_AWARE_LATEST_TRANSMIT_OFFSET,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_TRAFFIC_SPEC_TIME_AWARE_JITTER,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_NETWORK_REQUIREMENTS,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_NETWORK_REQUIREMENTS_NUM_SEAMLESS_TREES,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_TALKER_NETWORK_REQUIREMENTS_MAX_LATENCY,
+
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_LISTENER,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_LISTENER_NETWORK_REQUIREMENTS,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_LISTENER_NETWORK_REQUIREMENTS_NUM_SEAMLESS_TREES,
+  ELEMENT_KIND_DEPLOYMENT_CONF_TSN_LISTENER_NETWORK_REQUIREMENTS_MAX_LATENCY
+};
+
+enum element_data_type {
+  ELEMENT_DATA_TYPE_GENERIC,
+  ELEMENT_DATA_TYPE_DURATION
+};
+
+typedef int (* init_fn) (struct parse_sysdef_state * const pstate, struct xml_element *element);
+typedef void (* fini_fn) (struct xml_element *element);
+
+struct xml_element
+{
+  struct xml_element *parent;
+  enum element_kind kind;
+  enum element_data_type data_type;
+  bool retain;
+  bool handle_close;
+  struct xml_element *next;
+  fini_fn fini;
+};
+
+/* Type library */
+struct dds_sysdef_type {
+  struct xml_element xmlnode;
+  char *name;
+  unsigned char *identifier;
+  struct dds_sysdef_type_lib *parent;
+};
+
+struct dds_sysdef_type_lib {
+  struct xml_element xmlnode;
+  struct dds_sysdef_type *types;
+};
+
+/* QoS library */
+enum dds_sysdef_qos_kind {
+  DDS_SYSDEF_TOPIC_QOS,
+  DDS_SYSDEF_READER_QOS,
+  DDS_SYSDEF_WRITER_QOS,
+  DDS_SYSDEF_SUBSCRIBER_QOS,
+  DDS_SYSDEF_PUBLISHER_QOS,
+  DDS_SYSDEF_PARTICIPANT_QOS
+};
+
+#define QOS_POLICY_SYSDEF_STRUCT(p,t) \
+  struct dds_sysdef_ ## p { \
+    struct xml_element xmlnode; \
+    t values; \
+    uint32_t populated; \
+  };
+
+#define QOS_POLICY_DEADLINE_PARAM_PERIOD 1 << 0u
```

Comment:
```suggestion
#define QOS_POLICY_DEADLINE_PARAM_PERIOD (1 << 0u)
```
(not only for this one, obviously)

If I am not overlooking anything, it is today only used for `QOS_POLICY_DEADLINE_PARAMS` (and analogously for all other cases) and in that context the parentheses won't make any difference, but it is a common trap and we should not set it.